cloudapi/logs.go

```diff
@@ -176,3 +228,69 @@ func (c *Config) StreamLogsToLogger(
 		}
 	}
 }
+
+// timstampTrack is a safe-concurrent tracker
+// of the latest/most recent seen timestamp value.
+type timestampTrack struct {
+	// ts is timestamp in unix nano format
+	ts int64
+}
+
+// TimeOrNow returns as Time the latest tracked value plus 1ms
+// or Now as the default value.
+func (tst *timestampTrack) TimeOrNow() (t time.Time) {
+	t = time.Now()
+	if ts := atomic.LoadInt64(&tst.ts); ts > 0 {
+		// add 1ms for avoid possible repetition
+		t = time.Unix(0, ts).Add(1 * time.Millisecond)
+	}
+	return
+}
+
+// Set sets the tracked timestamp value.
+func (tst *timestampTrack) Set(ts int64) {
+	if ts < 1 {
+		return
+	}
```

Comment:
I kind of feel this isn't needed ... if we are logging you have a `ts` bigger than 1 ;)